extensions/mapstore/src/main/java/com/hazelcast/mapstore/Queries.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.mapstore;
+
+import com.hazelcast.sql.SqlColumnMetadata;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.joining;
+
+class Queries {
+
+    private final String loadQuery;
+
+    private final Function<Integer, String> loadAllFactory;
+    private final Map<Integer, String> loadAllQueries = new ConcurrentHashMap<>();
+
+    private final String loadAllKeys;
+
+    private final String storeInsert;
+    private final String storeUpdate;
+    private final String delete;
+
+    private final Function<Integer, String> deleteAllFactory;
+    private final Map<Integer, String> deleteAllQueries = new ConcurrentHashMap<>();
+
+    Queries(String mapping, String idColumn, List<SqlColumnMetadata> columnMetadata) {
+        loadQuery = "SELECT * FROM \"" + mapping + "\" WHERE \"" + idColumn + "\" = ?";
+
+        loadAllFactory = n -> "SELECT * FROM \"" + mapping + "\" WHERE \"" + idColumn + "\" IN ("
+                + queryParams(n)
+                + ")";
+
+        loadAllKeys = "SELECT \"" + idColumn + "\" FROM \"" + mapping + "\"";
+
+        String columnNames = columnMetadata.stream()
+                                           .map(sqlColumnMetadata -> '\"' + sqlColumnMetadata.getName() + '\"')
+                                           .collect(joining(", "));
+
+        storeInsert = "INSERT INTO \"" + mapping + "\" ( " + columnNames + " ) VALUES (" +
+                queryParams(columnMetadata.size()) +
+                ")";
+
+        String setClause = columnMetadata.stream()
+                                         .filter(cm -> !idColumn.equals(cm.getName()))
+                                         .map(cm -> '\"' + cm.getName() + "\" = ?")
+                                         .collect(joining(", "));
+
+        storeUpdate = "UPDATE \"" + mapping + "\" SET " + setClause
+                + " WHERE \"" + idColumn + "\" = ?";
+
+        delete = "DELETE FROM \"" + mapping + "\" WHERE \"" + idColumn + "\" = ?";
+        deleteAllFactory = n -> "DELETE FROM \"" + mapping + "\" WHERE \"" + idColumn + "\" IN ("
+                + queryParams(n)
+                + ")";
+    }
+
+    private String queryParams(long n) {
+        return Stream.generate(() -> "?").limit(n).collect(joining(", "));
+    }
+
+    String load() {
+        return loadQuery;
+    }
+
+    String loadAll(int n) {
+        return loadAllQueries.computeIfAbsent(n, loadAllFactory);
```

Comment:
No risk of a memory leak in case of multiple values of n? Do we need `loadAllQueries` map?

Reply:
The `n` is limited by this property `https://github.com/hazelcast/hazelcast/blob/deaeb753aa41ecf054f5d491dec6c0a75cb59756/hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java#L1152-L1153` so we may have up to `1000`  loadAll queries in the default configuration.